text/0000-north-star.md

```diff
@@ -0,0 +1,466 @@
+a public commitment to fixing them. They should be informed in part by inputs
+like [the survey] and [production user outreach], as well as an open discussion
+process. And while the end-product is problem-focused, the discussion is likely
+to touch on possible solutions as well. We shouldn't blindly commit to solving a
+problem without some sense for the plausibility of a solution in terms of both
+design and resources.
+
+[the survey]: https://blog.rust-lang.org/2016/06/30/State-of-Rust-Survey-2016.html
+[production user outreach]: https://internals.rust-lang.org/t/production-user-research-summary/2530
+
+Problem statements consist of a single sentence summarizing the problem, and one
+or more paragraphs describing it (and its importance!) in detail. Examples of
+good problem statements might be:
+
+- The Rust compiler is too slow for a tight edit-compile-test cycle
+- Rust lacks world-class IDE support
+- The Rust story for asynchronous I/O is very primitive
+- Rust compiler errors are dificult to understand
+- Rust plugins have no clear path to stabilization
+- Rust doesn't integrate well with garbage collectors
+- Rust's trait system doesn't fully support zero-cost abstractions
+- The Rust community is insufficiently diverse
+- Rust needs more training materials
+- Rust's CI infrastructure is unstable
+- It's too hard to obtain Rust for the platforms people want to target
+
+During the actual process each of these would be accompanied by a paragraph or
+more of justification.
+
+We strictly limit the planning phase to one month in order to keep the
+discussion focused and to avoid unrestrained bikeshedding. The activities
+specified here are not the focus of the project and we need to get through them
+efficiently and get on with the actual work.
+
+The core team is responsible for initiating the process, either on the internals
+forum or directly on the RFC repository, and the core team is responsible for
+merging the final RFC, thus it will be their responsibility to ensure that the
+discussion drives to a reasonable conclusion in time for the deadline.
+
+Once the year's problem statements are decided, a metabug is created for each on
+the rust-lang/rust issue tracker and tagged `R-problem-statement`. In the OP of
+each metabug the teams are responsible for maintaining a list of their goals,
+linking to tracking issues.
+
+## The little planning cycle (goals and tracking progress)
+
+The little cycle is where the solutions take shape and are carried out. They
+last one quarter - 3 months - and are the responsibility of individual teams.
+
+Each cycle the teams will have one week to update their set of _goals_. This
+includes both creating new goals and reviewing and revising existing goals. A
+goal describes a task that contributes to solving the year's problems. It may or
+may not involve a concrete deliverable, and it may be in turn subdivided into
+further goals.
```

Comment:
I much prefer a model where the teams update goals throughout the quarter and then have a week to reflect and confirm the goals. This allows more flexibility and less artificial time-boxing.